mapsforge-map/src/main/java/org/mapsforge/map/layer/overlay/Polygon.java

```diff
@@ -81,13 +85,44 @@ public synchronized void addPoints(List<LatLong> points) {
         updatePoints();
     }
 
+    /**
+     * Adds a hole to this polygon.
+     *
+     * A polygon can contain multiple holes. Each hole must be completely inside the polygon.
+     * Holes may not overlap. All hole's winding must be the opposite of the winding of the outer polygon
+     * (e.g. if outer polygon's points are in counter-clockwise order, then the points of each hole must be in
+     * clockwise order; and vice versa).
+     *
+     * @param holePoints path of the hole. If first and last point are not equal, path will be closed automatically
+     */
+    public synchronized void addHole(List<LatLong> holePoints) {
+        if (this.holes == null) {
+            this.holes = new CopyOnWriteArrayList<>();
+        }
+        this.holes.add(new CopyOnWriteArrayList<>(holePoints));
+    }
+
     public synchronized void clear() {
         this.latLongs.clear();
+        if (this.holes != null) {
+            this.holes.clear();
+        }
         updatePoints();
     }
 
     public synchronized boolean contains(LatLong tapLatLong) {
-        return LatLongUtils.contains(latLongs, tapLatLong);
+        final boolean contains = LatLongUtils.contains(latLongs, tapLatLong);
+        if (holes == null || !contains) {
+            return contains;
+        }
+
+        for(List<LatLong> hole : this.holes) {
```

Comment:
Can you please check the formatting here?

Reply:
I was not sure what formatting you are referring to. I inserted a blank now after the "for". Is this what you meant? (Are there any formatting guidelines for mapsforge code which I missed?)

Reply:
> I inserted a blank now after the "for".

:+1: